src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.SingleTC.cs

```diff
@@ -15,38 +15,42 @@ internal static partial class ValueSetFactory
     {
         private struct SingleTC : FloatingTC<float>, INumericTC<float>
         {
-            float INumericTC<float>.MinValue => float.MinValue;
+            float INumericTC<float>.MinValue => float.NegativeInfinity;
 
-            float INumericTC<float>.MaxValue => float.MaxValue;
+            float INumericTC<float>.MaxValue => float.PositiveInfinity;
 
             float FloatingTC<float>.NaN => float.NaN;
 
-            float FloatingTC<float>.MinusInf => float.NegativeInfinity;
-
-            float FloatingTC<float>.PlusInf => float.PositiveInfinity;
-
             /// <summary>
             /// The implementation of Next depends critically on the internal representation of an IEEE floating-point
             /// number.  Every bit sequence between the representation of 0 and MaxValue represents a distinct
             /// value, and the integer representations are ordered by value the same as the floating-point numbers they represent.
             /// </summary>
-            float INumericTC<float>.Next(float value)
+            public float Next(float value)
             {
                 Debug.Assert(!float.IsNaN(value));
-                Debug.Assert(!float.IsInfinity(value));
-                Debug.Assert(value != float.MaxValue);
+                Debug.Assert(value != float.PositiveInfinity);
+
+                if (value == 0)
+                    return float.Epsilon;
                 if (value < 0)
                 {
                     if (value == -float.Epsilon)
                         return 0.0f; // skip negative zero
+                    if (value == float.NegativeInfinity)
+                        return float.MinValue;
                     return -UintAsFloat(FloatAsUint(-value) - 1);
                 }
+                if (value == float.MaxValue)
+                    return float.PositiveInfinity;
 
                 return UintAsFloat(FloatAsUint(value) + 1);
             }
 
             private unsafe static uint FloatAsUint(float d)
             {
+                if (d == 0)
+                    return 0;
                 float* dp = &d;
                 uint* lp = (uint*)dp;
                 return *lp;
```

Comment:
Does this produce a different result than `return (uint)d;`. Similar question for conversion below.

Reply:
Yes.  This gives the bits of the value viewed as an integer.  It doesn't change the representation, just how it is viewed.

---
In reply to: [401126937](https://github.com/dotnet/roslyn/pull/42886#discussion_r401126937) [](ancestors = 401126937)